hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/wal/CompressionContext.java

```diff
@@ -70,18 +150,55 @@ public CompressionContext(Class<? extends Dictionary> dictType, boolean recovere
     if (hasTagCompression) {
       tagCompressionContext = new TagCompressionContext(dictType, Short.MAX_VALUE);
     }
+    if (hasValueCompression && valueCompressionType != null) {
+      valueCompressor = new ValueCompressor(valueCompressionType);
+    }
+  }
+
+  public CompressionContext(Class<? extends Dictionary> dictType, boolean recoveredEdits,
+      boolean hasTagCompression)
+      throws SecurityException, NoSuchMethodException, InstantiationException,
+        IllegalAccessException, InvocationTargetException, IOException {
+    this(dictType, recoveredEdits, hasTagCompression, false, null);
+  }
+
+  public boolean hasTagCompression() {
+    return tagCompressionContext != null;
+  }
+
+  public boolean hasValueCompression() {
+    return valueCompressor != null;
   }
 
-  public Dictionary getDictionary(Enum dictIndex) {
+  public Dictionary getDictionary(Enum<DictionaryIndex> dictIndex) {
```

Comment:
Drop `Enum<...>` and make this simply `Dictionary getDictionary(DictionaryIndex dictIndex)`

Reply:
I cannot do this without a ripple effect through WALCodec, which is LimitedPrivate.

Reply:
Let me just leave this alone, the same as before I touched this file.